chainstate/src/detail/tokens/mod.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{
+    chain::{
+        tokens::{TokenData, TokensError},
+        Block, ChainConfig, Transaction,
+    },
+    primitives::{Amount, Id, Idable},
+};
+use utils::ensure;
+
+pub fn check_tokens_transfer_data(
+    source_block_id: Id<Block>,
+    tx: &Transaction,
+    amount: &Amount,
+) -> Result<(), TokensError> {
+    // Check amount
+    ensure!(
+        amount > &Amount::from_atoms(0),
+        TokensError::TransferZeroTokens(tx.get_id(), source_block_id)
+    );
+
+    Ok(())
+}
+
+pub fn check_tokens_burn_data(
```

Comment:
What does `data` have to do with this?

Reply:
Don't ask for explanation... just make suggestions (and I genuinely mean that). That's the shortest path. Let's try to get this done with.

Reply:
We literally have the type `TokenData`, that essentially enum with burn, transfer, and issuance data for output.